### ISSUE TITLE
Checkbox label position 1px/2px too low

### ISSUE THREAD

**Issue by pudjeeb** [adobe/spectrum-css#406]
## Description
Checkbox label is 1px (desktop) / 2px (mobile) lower than the design

## Steps to reproduce

1. Go to https://opensource.adobe.com/spectrum-css/components/checkbox/
2. Screenshot checkbox
3. Compare with UI kit

## Expected behavior
Label should be 1px higher (desktop) or 2px higher (mobile)

## Screenshots
(red is design)
desktop:
<img width="545" alt="checkbox-desktop" src="https://user-images.githubusercontent.com/5943545/68907439-a4861f80-06fc-11ea-8fca-3d0a87318d60.png">

mobile:
<img width="487" alt="checkbox-mobile" src="https://user-images.githubusercontent.com/5943545/68907441-a51eb600-06fc-11ea-9e75-f72817c1b8c5.png">

## Environment
 - **Spectrum CSS version:** 
 - **Browser(s) and OS(s):** 

## Additional context
Came across while doing React review. Not sure if this is some rendering issue or not :( sorry if this is not relevant!

**Comment by devongovett** [comment]
FYI, looks even more off in Firefox and Edge on both macOS and Windows. More like it's bottom aligned. Here's a FF mac screenshot:

![image](https://user-images.githubusercontent.com/19409/70278755-b7bf6480-1769-11ea-98f6-41f7a124bdc7.png)

**Comment by bernhard-adobe** [comment]
thanks @devongovett for the feedback.
We did a deeper dive into this and it turns out this 1px rendering bug is coming from "browser jiggering" based on the rendering engine. 

**TLDR;**  different rendering engines place the baseline differently.


I compared a couple of browsers and I might have a potential solution in this commit: https://github.com/adobe/spectrum-css/pull/412/commits/2d6b43ae1a1dca22992df4e1a98af7c6ebbdd3be

Here is a comparison on how a couple of evergreen browser might render it differently. We may have to discuss if it is worth it using a "1.49" line-height fix. A couple of solutions point into the direction of just letting this be. Maybe  @NateBaldwinDesign  might have some good feedback for this situation.

Screenshot:

![line-height-rendering-bugs](https://user-images.githubusercontent.com/52184321/70492989-24fd2d80-1abb-11ea-836a-2b800f2b66cb.png)

**Comment by lazd** [comment]
@bernhard-adobe so it sounds like setting `line-height: 1.49` is a hack to fix the issue in Firefox and Edge, and without it, the label is 1px too low. Personally, I say that's an acceptable solution and I would accept the PR with that change. 

@GarthDB and @jianliao, your thoughts? Let's get feedback on this by EOD so we can merge.

**Comment by NateBaldwinDesign** [comment]
@bernhard-adobe this is pretty crazy. **I am fine with changing the line height in those browsers to account for the problem this causes**.

Considering this is a browser-specific rendering issue w/r/t line heights, I assume this means all components in Firefox and Edge would have this same problem? If so, it seems this is more than just a Checkbox-related issue. 

Here's where my head is at on this -> For checkbox and all components whose type is derived from `Body`, the line-height is defined at 1.5x the font size in Latin scripts. A few questions that arise in my mind are this:
1. Would it be better to provide `line-heights` as multiplier (unitless) values rather than their resolved pixel value? (ie `1.3`, `1.5`, `1.7`)? Would this provide any help regarding these rendering issues?
2. If this is a universal rendering issue and appears on other type sizes, is there a more broad rule we should apply such that for Firefox & Edge, all line heights = `line-height - 0.01`?

**Comment by bernhard-adobe** [comment]
correct, this might appear in other components as well. 
To answer your questions:

1. Which multiplier would you like to use?
2. I tried that with  calc but the line-heights I tried it with were PX based and that didn't fix the issue. Using the resolved pixel values worked only in this scenario.

**Comment by devongovett** [comment]
That is super strange. Does that solution work at different scales too (e.g. large scale)? What if you zoom in to > 100%? I'd be wary of setting exact numeric values like that...

**Comment by devongovett** [comment]
![image](https://user-images.githubusercontent.com/19409/71047240-fdd0dc80-2108-11ea-9883-6d2f8bd0debd.png)
@bernhard-adobe Still appears too low to me on master...

**Comment by bernhard-adobe** [comment]
Thanks for providing the screenshot @devongovett . 
May I ask what browser you are using, if your screen is a retina-based screen and the name of the operating system?

I have ran the tests again on macOS Mojave 10.14.5 (18F203) on a non-retina Apple Thunderbolt screen 2011

Master at: https://github.com/adobe/spectrum-css/tree/4747bc456053a7ad5bb84a29385f5d00a6029681

Browsers from left to right:
1. Google Chrome Canary Version 81.0.3999.0 (Official Build) canary (64-bit)
2. Firefox Version 71.0 (64-bit)
3. Safari Version 12.1.1 (14607.2.6.1.1)
4. Microsoft Edge on macOS Version 79.0.309.47 (Official build) Beta (64-bit)
5. Microsoft Edge Version 42.17134.1038.0  in a VMware Fusion on Windows 10 1803, Microsoft EdgeHTML 17.17134

Here the screenshots: notice the correct gap of 1px between the bottom of the checkbox and the label. 

![test-master-2019-12-18](https://user-images.githubusercontent.com/52184321/71083837-3ff33000-2194-11ea-9aba-ef74267ca88e.png)

The only difference I noticed is in Edge in VMware Fusion. A very subtle anti-aliasing can be seen at the bottom of the label. This however could be a false positive as I am running this in a virtual machine on my mac and VMware fusion might scale the DPI rate of my Thunderbolt screen wrong towards Windows. There are a couple of settings in VMware Fusion that allow further customization.

**Comment by devongovett** [comment]
Those all look baseline aligned to me, not centered. You're seeing the same as me.

**Comment by dannify** [comment]
We have made this adjustment in react-spectrum to vertically align the labels for checkbox/radio/status light/switch.
https://github.com/adobe/react-spectrum-v3/pull/138
This should be discussed in the new year.

**Comment by bernhard-adobe** [comment]
OH I see the issue. @lazd  and I have worked to align those items constantly on the baseline as we thought that was the correct look and feeling. double checking in our Spectrum Specs https://spectrum.corp.adobe.com/page/radio-button/ it appears that those labels are a bit higher than than we expected it. I have correct that this in #458
